assets/app/scripts/directives/logViewer.js

```diff
@@ -224,6 +226,25 @@ angular.module('openshiftConsole')
               goChromeless: logLinks.chromelessLink,
               restartLogs: streamLogs
             });
+
+            APIDiscovery
+              .getLoggingURL()
+              .then(function(url) {
+                var hasComponents = $scope.context.project.metadata.name && $scope.name && url;
```

Comment:
If project.metadata.name is undefined, I'd expect project.metadata to be undefined and a runtime error.

Do we need to be watching these properties if we expect any of them to be null on initialization?